src/compiler/transformers/es6.ts

```diff
@@ -640,13 +644,13 @@ namespace ts {
          * @param node A ClassExpression or ClassDeclaration node.
          * @param hasExtendsClause A value indicating whether the class has an `extends` clause.
          */
-        function transformClassBody(node: ClassExpression | ClassDeclaration, hasExtendsClause: boolean): Block {
+        function transformClassBody(node: ClassExpression | ClassDeclaration, hasExtendsClause: boolean, closingBraceLocation: TextRange): Block {
             const statements: Statement[] = [];
             startLexicalEnvironment();
             addExtendsHelperIfNeeded(statements, node, hasExtendsClause);
             addConstructor(statements, node, hasExtendsClause);
             addClassMembers(statements, node);
-            statements.push(createReturn(getDeclarationName(node)));
+            statements.push(createReturn(getDeclarationName(node), /*location*/ closingBraceLocation));

```

Comment:
Just making sure. the emitted return statement will be mapped to the closing-brace in the original code?